### PR TITLE
Add bytea to working with types and improve img example

### LINKED ISSUE #539: Use the raw-media-types config in the images how-to
It was suggested here https://gitter.im/begriffs/postgrest?at=6285fe78c61b987d33b763fe that the `raw-media-types` could be used to replace the minimal example on how to serve images instead of using Nginx rewrite rules.

Perhaps see which image types are common for popular browsers and add that to `raw-media-types`, for instance.

May be related to #327.

### PR DESCRIPTION
**Pull request by laurenceisla**
Closes #539 and adds info mentioned in https://github.com/PostgREST/postgrest-docs/issues/539#issuecomment-1136213676

### CODE REVIEW

File: docs/how-tos/working-with-postgresql-data-types.rst [hunk 1]
```diff
@@ -5,6 +5,10 @@ Working with PostgreSQL data types
 
 PostgREST makes use of PostgreSQL string representations to work with data types. Thanks to this, you can use special values, such as ``now`` for timestamps, ``yes`` for booleans or time values including the time zones. This page describes how you can take advantage of these string representations to perform operations on different PostgreSQL data types.
 
+.. contents::
+  :local:
+  :depth: 1
+
```
Comment (steve-chavez) [inline docs/how-tos/working-with-postgresql-data-types.rst:11]: Nice! I thought the TOC needed to be manual.

File: docs/how-tos/working-with-postgresql-data-types.rst [hunk 2]
```diff
@@ -461,6 +465,80 @@ Finally, do the request :ref:`casting the range column <casting_columns>`:
...
+   create table files (
+     id serial primary key,
+     file bytea
+   );
+
+   create function upload_binary(bytea) returns void as $$
+     insert into files (file) values ($1);
+   $$ language sql;
+
+.. TODO: GRANT FOR SERIAL
```
Comment (steve-chavez) [inline docs/how-tos/working-with-postgresql-data-types.rst:484]: You can use the `generated always as identity` instead of `serial`.

Comment (laurenceisla) [reply docs/how-tos/working-with-postgresql-data-types.rst:484]: Didn't know that it's even suggested not to use `serial`: https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_serial

File: docs/how-tos/providing-images-for-img.rst [hunk 3]
```diff
@@ -36,64 +25,40 @@ First, we need a public table for storing the files.
...
-       return 404;
-       # request binary output
-       proxy_set_header Accept application/octet-stream;
-       # usual proxy setup
-       proxy_hide_header Content-Location;
-       add_header Content-Location /api/$upstream_http_content_location;
-       proxy_set_header  Connection "";
-       proxy_http_version 1.1;
-       proxy_pass http://localhost:3000/;
-     }
```
Comment (laurenceisla) [inline docs/how-tos/providing-images-for-img.rst:64]: It looks like the examples do not need the reverse proxy anymore. Maybe add another section below, mentioning that `raw-media-types` won't be needed if the :ref:`Accept: application/octet-stream` header is added in Nginx?

Comment (steve-chavez) [reply docs/how-tos/providing-images-for-img.rst:64]: Oh, so we don't need Nginx at all now right? Looks much better this way.

> that raw-media-types won't be needed if the Accept: application/octet-stream header is added in Nginx?

I think, it's fine to leave Nginx out for this use case.

File: docs/how-tos/providing-images-for-img.rst [hunk 4]
```diff
@@ -5,25 +5,14 @@ Providing images for ``<img>``
...
 
-In this how-to, you will learn how to create an endpoint for providing images to HTML :code:`<img>` tags without client side JavaScript.
-The resulting HTML might look like this:
-
-.. code-block:: html
-
-   <img src="http://host/files/42/cats.jpeg" alt="Cute Kittens"/>
-
-In fact, the presented technique is suitable for providing not only images, but arbitrary files.
+In this how-to, you will learn how to create an endpoint for providing images to HTML :code:`<img>` tags without client side JavaScript. In fact, the presented technique is suitable for providing not only images, but arbitrary files.
```
Comment (steve-chavez) [inline docs/how-tos/providing-images-for-img.rst:8]: I'd like to add a warning for this how-to, what I mentioned in https://github.com/PostgREST/postgrest-docs/issues/327#issue-626719955.

> Also maybe add a note with a link to https://wiki.postgresql.org/wiki/BinaryFilesInDB
> And mention CDN as an alternative for scaling, see the discussion on https://github.com/PostgREST/postgrest/issues/1077.

Maybe the CDN part can be left out, it's not only about scaling in throughput but pg WALs(backups, replicas) also get bloated with big files.

Comment (steve-chavez) [reply docs/how-tos/providing-images-for-img.rst:8]: Just linking to the discussion would be good though

File: docs/how-tos/providing-images-for-img.rst [hunk 5]
```diff
@@ -5,25 +5,18 @@ Providing images for ``<img>``
...
-   <img src="http://host/files/42/cats.jpeg" alt="Cute Kittens"/>
+We will start with a minimal example that highlights the general concept.
+Afterwards we present a more detailed solution that fixes a few shortcomings of the first approach.
 
-In fact, the presented technique is suitable for providing not only images, but arbitrary files.
+.. note::
 
-We will start with a minimal example that highlights the general concept.
-Afterwards we present are more detailed solution that fixes a few shortcomings of the first approach.
+   There are some caveats when saving images in the database as `mentioned in the PostgreSQL wiki <https://wiki.postgresql.org/wiki/BinaryFilesInDB>`_. Also, consider using a Content Delivery Network (CDN) for scalability purposes, see `this discussion <https://github.com/PostgREST/postgrest/issues/1077>`_ for reference.
```
Comment (steve-chavez) [inline docs/how-tos/providing-images-for-img.rst:15]: Can't think of good words now, but this needs to be more than "some caveats", almost like "think twice before doing this". Also, the CDN thing should be left out(I don't know how good it is to recommend it).

**Review by steve-chavez — APPROVED** [review]

File: docs/how-tos/working-with-postgresql-data-types.rst [hunk 6]
```diff
@@ -461,6 +465,80 @@ Finally, do the request :ref:`casting the range column <casting_columns>`:
 
       create cast (mytsrange as json) with function mytsrange_to_json(mytsrange) as assignment;
 
+Binary Data Types
+-----------------
```
Comment (steve-chavez) [inline docs/how-tos/working-with-postgresql-data-types.rst:469]: How about just bytea?
```suggestion
Bytea
--------
```

File: docs/how-tos/working-with-postgresql-data-types.rst [hunk 7]
```diff
@@ -461,6 +465,80 @@ Finally, do the request :ref:`casting the range column <casting_columns>`:
...
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/files?select=file&id=eq.1" \
+      -H "Accept: image/png"
+
+See :ref:`providing_img` for a step-by-step example on how to handle images in HTML.
+
+.. note::
+
+   Be careful when saving binaries in the database, you may want to use a separate storage service to handle files. See `the PostgreSQL wiki <https://wiki.postgresql.org/wiki/BinaryFilesInDB>`_ for reference.
```
Comment (steve-chavez) [inline docs/how-tos/working-with-postgresql-data-types.rst:540]:
```suggestion
   Be careful when saving binaries in the database, having a separate storage service for these is preferable in most cases. See `the PostgreSQL wiki <https://wiki.postgresql.org/wiki/BinaryFilesInDB>`_ for reference.
```

File: docs/how-tos/working-with-postgresql-data-types.rst [hunk 8]
```diff
@@ -461,6 +465,80 @@ Finally, do the request :ref:`casting the range column <casting_columns>`:
...
+    Accept: image/png
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/files?select=file&id=eq.1" \
+      -H "Accept: image/png"
+
+See :ref:`providing_img` for a step-by-step example on how to handle images in HTML.
+
+.. note::
```
Comment (steve-chavez) [inline docs/how-tos/working-with-postgresql-data-types.rst:538]:
```suggestion
.. warning::
```

**Pull request merged**